src/YesSql.Core/Services/DbBlockIdGenerator.cs

```diff
@@ -43,9 +43,9 @@ public async Task InitializeAsync(IStore store, ISchemaBuilder builder)
             _tablePrefix = store.Configuration.TablePrefix;
             _store = store;
 
-            SelectCommand = "SELECT " + _dialect.QuoteForColumnName("nextval") + " FROM " + _dialect.QuoteForTableName(_tablePrefix + TableName) + " WHERE " + _dialect.QuoteForTableName("dimension") + " = @dimension;";
-            UpdateCommand = "UPDATE " + _dialect.QuoteForTableName(_tablePrefix + TableName) + " SET " + _dialect.QuoteForColumnName("nextval") + "=@new WHERE " + _dialect.QuoteForColumnName("nextval") + " = @previous AND " + _dialect.QuoteForColumnName("dimension") + " = @dimension;";
-            InsertCommand = "INSERT INTO " + _dialect.QuoteForTableName(_tablePrefix + TableName) + " (" + _dialect.QuoteForColumnName("dimension") + ", " + _dialect.QuoteForColumnName("nextval") + ") VALUES(@dimension, @nextval);";
+            SelectCommand = "SELECT " + _dialect.QuoteForColumnName("nextval") + " FROM " + _dialect.QuoteForTableName(_tablePrefix + store.Configuration.NamingPolicy.ConvertName(TableName)) + " WHERE " + _dialect.QuoteForColumnName("dimension") + " = @dimension;";
+            UpdateCommand = "UPDATE " + _dialect.QuoteForTableName(_tablePrefix + store.Configuration.NamingPolicy.ConvertName(TableName)) + " SET " + _dialect.QuoteForColumnName("nextval") + "=@new WHERE " + _dialect.QuoteForColumnName("nextval") + " = @previous AND " + _dialect.QuoteForColumnName("dimension") + " = @dimension;";
```

Comment:
If the naming policy was part of the dialect, then we wouldn't have to change most code as it would be automatic when QuoteForXYZ is called.

But that would mean to have an overload with QuoteForTableName(string prefix, string name) so we can only convert the table name and not the prefix.

Reply:
I was thinking about that before, but these methods are `abstract`, so let me revise the code and see what should I come up with

Reply:
> But that would mean to have an overload with QuoteForTableName(string prefix, string name) so we can only convert the table name and not the prefix.

I realize that there 're some places will not work, on example `CreateTableCommand` because it pass a table name with a prefix to the dialect